Project/Galagram/ViewModel/Commands/Admin/MultipleCommand.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System.Windows.Input;
+
+namespace Galagram.ViewModel.Commands.Admin
+{
+    /// <summary>
+    /// Represents multiple command
+    /// </summary>
+    public class MultipleCommand : CommandBase
+    {
+        // FIELDS
+        ICommand[] commands;
+
+        // EVENT
+        /// <summary>
+        /// Occurs when state of the command has been changed
+        /// </summary>
+        public override event System.EventHandler CanExecuteChanged
+        {
+            add
+            {
+                CommandManager.RequerySuggested += value;
+            }
+            remove
+            {
+                CommandManager.RequerySuggested -= value;
+            }
+        }
+
+        // CONSTRUCTORS
+        /// <summary>
+        /// Initialize a new instance of <see cref="MultipleCommand"/>
+        /// </summary>
+        /// <param name="commands">
+        /// The array of <see cref="ICommand"/>
+        /// </param>
+        public MultipleCommand(ICommand[] commands)
+        {
+            this.commands = commands;
+        }
+
+        // METHODS
+        /// <summary>
+        /// Checks if command can be executed
+        /// </summary>
+        /// <param name="parameter">
+        /// Additionals parameters
+        /// </param>
+        /// <returns>
+        /// True if command can be executed, otherwise — false
+        /// </returns>
+        public override bool CanExecute(object parameter)
+        {
+            Core.Logger.GetLogger.LogAsync(Core.LogMode.Debug, $"Can execute {nameof(MultipleCommand)}");
+
+            // check if all commands can be executed
+            bool canExecute = true;
+            foreach (ICommand command in commands)
+            {
+                canExecute &= command.CanExecute(parameter);
```

Comment:
there is no needs to check each command. you can interrupt loop with first false occurence

Reply:
yes